lib: location: support external transport method for P-GPS data (#7138)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

      * :c:member:`get` callback in :c:struct:`bt_mesh_sensor_series`.
      * :c:member:`get` callback in :c:struct:`bt_mesh_sensor`.

Modem libraries
------------------------

* :ref:`lib_location` library:

  * Added:

>     * Support for P-GPS data retrieval from an external source, which the application implements separately.

Reviewer: ```suggestion
    * Support for P-GPS data retrieval from an external source, implemented separately by the application.
```

>       Enabled by setting :kconfig:`CONFIG_LOCATION_METHOD_GNSS_PGPS_EXTERNAL`.

Reviewer: ```suggestion
      Enabled by setting the :kconfig:`CONFIG_LOCATION_METHOD_GNSS_PGPS_EXTERNAL` option.
```

>       Location library throws event :c:enum:`LOCATION_EVT_GNSS_PREDICTION_REQUEST` when assistance is needed.

Reviewer: ```suggestion
      The library triggers a :c:enum:`LOCATION_EVT_GNSS_PREDICTION_REQUEST` event when assistance is needed.
```

      * :c:member:`get` callback in :c:struct:`bt_mesh_sensor_series`.
      * :c:member:`get` callback in :c:struct:`bt_mesh_sensor`.

Modem libraries
------------------------

* :ref:`lib_location` library:

  * Added:

    * Support for P-GPS data retrieval from an external source, implemented separately by the application.

>       Enabled by setting the :kconfig:`CONFIG_LOCATION_METHOD_GNSS_PGPS_EXTERNAL` option.

Reviewer: ```suggestion
      Enabled by setting the :kconfig:option:`CONFIG_LOCATION_METHOD_GNSS_PGPS_EXTERNAL` option.
```
